src/main/java/click/escuela/school/admin/service/impl/TeacherServiceImpl.java

```diff
@@ -48,4 +48,18 @@ public void update(TeacherApi teacherApi) throws TransactionException {
 						TeacherMessage.GET_ERROR.getDescription())));
 	}
 
+	public TeacherDTO getById(String id) throws TransactionException {
+		Teacher teacher = findById(id).orElseThrow(() -> new TransactionException(TeacherMessage.GET_ERROR.getCode(),
+				TeacherMessage.GET_ERROR.getDescription()));
+		return Mapper.mapperToTeacherDTO(teacher);
+	}
+
+	public List<TeacherDTO> getBySchoolId(String schoolId) {
+		return Mapper.mapperToTeachersDTO(teacherRepository.findBySchoolId(Integer.valueOf(schoolId)));
+	}
+
+	public List<TeacherDTO> getByCourseId(String courseId) {
```

Comment:
if fail teacherRepository we should catch this error and throw an TransactionException with a nicer message